pkg/ext/stores/tokens/tokens.go

```diff
@@ -613,10 +621,15 @@ func (t *SystemStore) Create(ctx context.Context, group schema.GroupResource, to
 		return nil, apierrors.NewBadRequest("operation references a disabled user")
 	}
 
+	authTokenID, err := t.auth.SessionID(ctx)
+	if err != nil {
+		return nil, apierrors.NewInternalError(fmt.Errorf("error getting the authentication token: %w", err))
+	}
+
 	// Get token of the request and use its principal as ours. Any attempt
 	// by the user to set their own information for the principal is
 	// discarded and written over. No checks are made, no errors are thrown.
-	requestToken, err := t.Fetch(t.auth.SessionID(ctx))
+	requestToken, err := t.Fetch(authTokenID)
```

Comment:
This only makes sense if requestToken is not empty. If it is we need to return an appropriate error (not a 500). E.g.
```
	if requestToken == "" {
		return nil, apierrors.NewForbidden(gvr.GroupResource(), "", fmt.Errorf("missing request token ID"))
	}
```